### ISSUE TITLE
Karma Tests & SourceMaps

### ISSUE THREAD

**Issue by bluegaspode** [jhipster/generator-jhipster#6400]
##### **Overview of the issue**
I'm facing the issue, that for some reason sourcemaps don't seem to work in Karma Unit tests.
I.e. a failed test points to line 'entry.ts:88282', which is not of much use.

```
PhantomJS 2.1.1 (Mac OS X 0.0.0) Component Tests Project Management Detail Component OnInit Should call load all on init FAILED
	Expected spy find to have been called with [ 123 ] but it was never called.
	http://localhost:9877spec/entry.ts:88282:58

```
##### **Motivation for or Use Case**

It should be easier to pin down errors, that are found by Karma tests.
Above errors are much harder to track, when you get javascript errors somewhere deep in the code and need to find the exact place of the error.

##### **Reproduce the error**

- create a fresh project with monolith-defaults ('jhipster')
- create an entity called 'Project'
- in ProjectDetailComponent.ngOnInit (project-detail.component.ts) comment out everything 
- run karma test (yarn run test) - it will show a failed test

##### **Related issues**
I haven't found any

##### **Suggest a Fix**

I searched a lot about this bug, and finally this comment 
https://github.com/webpack-contrib/karma-webpack/issues/109#issuecomment-224961264
helped to solve the problem.

I add this to webpack.test.js, removed "devtool: 'inline-source-map'" and finally have the following output with working sourcemap resolution: 

```
PhantomJS 2.1.1 (Mac OS X 0.0.0) Component Tests Project Management Detail Component OnInit Should call load all on init FAILED
	Expected spy find to have been called with [ 123 ] but it was never called.
	http://localhost:9877webpack:///src/test/javascript/spec/app/entities/project/project-detail.component.spec.ts:56:54 <- spec/entry.ts:88282:58
```

##### **JHipster Version(s)**
4.8.2
##### **JHipster configuration**
<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp"
    },
    "jhipsterVersion": "4.8.2",
    "baseName": "jhipster4",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "session",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "enableSwaggerCodegen": false,
    "rememberMeKey": "replaced-by-jhipster-info",
    "clientFramework": "angularX",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": false
  }
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Project (project) {
  name String required minlength(10) maxlength(50),
  number Integer required min(5) max(100),
  someDate LocalDate
}

paginate Project with pagination

</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_66"
Java(TM) SE Runtime Environment (build 1.8.0_66-b17)
Java HotSpot(TM) 64-Bit Server VM (build 25.66-b17, mixed mode)

git version 2.10.1 (Apple Git-78)

node: v6.10.0

npm: 3.9.2

bower: 1.7.9

gulp:
[16:27:45] CLI version 3.9.1

yarn: 0.27.5

Docker version 1.13.0, build 49bf474

docker-compose version 1.10.0, build 4bd6f1a

##### **Browsers and Operating System**
Mac OSX 10.11.6

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by jdubois** [comment]
Yes it makes sense: maybe we should have a specific config for tests.

**Comment by deepu105** [comment]
We already ahve specific webpack config for test. @bluegaspode would you like to PR the changes you made for this?

**Comment by jdubois** [comment]
Oh I missed this, thanks @deepu105 ! So yes +1

**Comment by deepu105** [comment]
fixed via #6407
